modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/asserts/CompletableFutureAssert.java

```diff
@@ -92,12 +92,18 @@ public static <X extends Throwable> X assertWillThrow(
         try {
             normalResult = future.get(timeout, timeUnit);
         } catch (TimeoutException e) {
-            return fail("Expected the future to be completed with an exception of class in time, but it did not");
+            if (expectedExceptionClass.isInstance(e)) {
```

Comment:
Why do we need a separate case for this exception at all?

Reply:
To tell that we failed because the future was not completed *in time*, and not just because it failed